eth/web3.c

```diff
@@ -235,3 +236,63 @@ int eth_getTransactionReceipt(web3_ctx_t *web3, const tx_hash_t *tx_hash)
     WEB3_TERMINATOR();
     return 0;
 }
+
+int eth_convert(const uint256_t *amount, enum ETH_UNIT from, enum ETH_UNIT to, char *buf, size_t buf_size)
+{
+    assert(amount != NULL);
+    uint256_t conversion = *amount;
+    uint256_t power;
+
+    // add zeros
+    if (from >= to) {
+        set256_uint64(&power, pow(10, (int) from - (int) to));
```

Comment:
you can't really use `pow()` here since embedded systems can't use floats

Reply:
Since the custom pow function can't be built without using `uint128_t` (2 ^ 64 < 30) I used simple string manipulation for the `eth_convert` function, it's more efficient.